### PR TITLE
545 Keep selected variable in var browser as filters are changed

### LINKED ISSUE #545: [Feature Request]: UX Preserve table selection on `tm_variable_browser`
### Feature description

The row selection in the variable browser is not preserved every time the Filters are changed.

It would be user-friendly to keep the selection state of the current row on the server side and keep that selection across updates to the underlying data.

### Example screencast

[Screencast from 2023-08-02 15-05-44.webm](https://github.com/insightsengineering/teal.modules.general/assets/211358/5bf184ab-e0f2-4a72-ab9c-56ac25f7cb3a)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.

### Contribution Guidelines

- [X] I agree to follow this project's Contribution Guidelines.

### Security Policy

- [X] I agree to follow this project's Security Policy.

### PR DESCRIPTION
**Pull request by averissimo**
# Pull Request

Fixes #545
Fixes https://github.com/insightsengineering/teal.gallery/issues/19

#### Changes description

- Restores the selection based on the selection _(using variable name instead of indexes)_
- Restores the page _(non-trivial as it needs to calculate the index of the first row in the table)_

#### TODO before ready to review:

- [x] Support selection on non-first page rows
  - uses `displayStart` to set the first row of the page of the selected item

### CODE REVIEW

File: R/tm_variable_browser.R [hunk 1]
```diff
@@ -1154,98 +1155,123 @@ render_tab_header <- function(dataset_name, output, data) {
...
+    selected_page_ix <- 0
 
-        # calculate number of missing values
-        missings <- vapply(
-          df,
-          var_missings_info,
-          FUN.VALUE = character(1),
-          USE.NAMES = FALSE
-        )
+    # Retrieve current selected variable if any
```
Comment (averissimo) [inline R/tm_variable_browser.R:1247]: Changes start here until line `1261`

File: R/tm_variable_browser.R [hunk 2]
```diff
@@ -1154,98 +1155,123 @@ render_tab_header <- function(dataset_name, output, data) {
 #' @param parent_dataname (`character`) the name of a parent `dataname` to filter out variables from
 #' @inheritParams render_tabset_panel_content
 #' @keywords internal
-render_tab_table <- function(dataset_name, parent_dataname, output, data, input, columns_names) {
+render_tab_table <- function(dataset_name, parent_dataname, output, data, input, columns_names, plot_var) {
```
Comment (gogonzo) [inline R/tm_variable_browser.R:1158]: Please add documentation of a new argument

Comment (averissimo) [reply R/tm_variable_browser.R:1158]: This variable's documentation is already defined and inherited from `render_tabset_panel_content` and is already updated on `man/render_tab_table.Rd`

File: R/tm_variable_browser.R [hunk 3]
```diff
@@ -1154,98 +1155,123 @@ render_tab_header <- function(dataset_name, output, data) {
...
+        unlist(
+          lapply(
+            df,
+            function(x) {
+              `if`(is.null(attr(x, "label")), "", attr(x, "label"))
+            }
+          )
+        ),
+        names(df)
+      )
```
Comment (gogonzo) [inline R/tm_variable_browser.R:1202]:
```suggestion
      labels <- teal.data::col_labels(df)
```

File: R/tm_variable_browser.R (hunk 2 above)
Comment (gogonzo) [reply R/tm_variable_browser.R:1158]: Thats fine. Sorry, didnt realize this

**Review by gogonzo — APPROVED** [review]
👍 works as expected

**Pull request merged**